examples/mixedelasticity_KSP/qfunctions/volumetric-geometry-3d.h

```diff
@@ -0,0 +1,44 @@
+/// @file
+/// volumetric geometric factors computation QFunction source
+
+#ifndef volumetric_geometry3d_qf_h
+#define volumetric_geometry3d_qf_h
+
+#include "utils.h"
+// -----------------------------------------------------------------------------
+// This QFunction setup q_data, the inverse of the Jacobian
+// Inputs:
+//   J          : dx/dX
+//   w          : weight of quadrature
+//
+// Output:
+//   q_data     : updated weight of quadrature and inverse of the Jacobian J; [wdetJ, dXdx]
+// -----------------------------------------------------------------------------
+CEED_QFUNCTION(SetupVolumeGeometry3D)(void *ctx, CeedInt Q, const CeedScalar *const *in, CeedScalar *const *out) {
+  // Inputs
+  const CeedScalar(*J)[3][CEED_Q_VLA] = (const CeedScalar(*)[3][CEED_Q_VLA])in[0];
+  const CeedScalar(*w)                = in[1];
+
+  // Outputs
+  CeedScalar(*q_data)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0];
+
+  // Quadrature Point Loop
+  CeedPragmaSIMD for (CeedInt i = 0; i < Q; i++) {
+    // Setup
+    const CeedScalar dxdX[3][3] = {
+        {J[0][0][i], J[1][0][i], J[2][0][i]},
+        {J[0][1][i], J[1][1][i], J[2][1][i]},
+        {J[0][2][i], J[1][2][i], J[2][2][i]}
+    };
+    const CeedScalar detJ = ComputeDet3(dxdX);
+    CeedScalar       dXdx_voigt[9];
+    MatComputeInverseNonSymmetric3(dxdX, detJ, dXdx_voigt);
+    q_data[0][i] = w[i] * detJ;
+    for (CeedInt j = 0; j < 9; j++) {
+      q_data[j + 1][i] = dXdx_voigt[j];
+    }
+  }  // End of Quadrature Point Loop
```

Comment:
@jedbrown @jeremylt I have one question here. Since the MMS elasticity doesn't give the correct order of convergence in Ratel, I copied BP4 from example/petsc then extended to linear elasticity and after that mixed-elasticity to make sure everything is correct. But instead of creating `q_data` as we have in libCEED/Ratel (by creating matrix A[3][3], ...) I tried to use `MatComputeInverseNonSymmetric3` function which I tested for random 3x3 matrix in a notebook.
This setup gives exactly the same L2 error/number of iteration/convergence order as we have in `example/petsc/bps` (bp4) but only for `hex` element not for `tet`, why?
And the only difference is how I created `q_data`, is it wrong to setup `dxdX` like above for `tet` element?

Reply:
I made a change in Ratel yesterday that may have fixed the order of convergence

Reply:
No ratel still doesn't give the correct order. My question here is about bp4/tet element. Why when I setup q_data like above works for `hex` but not `tet`?

Reply:
Elasticity example in this branch works fine, you can run by `./conv_test.sh -o 1` (or 2,3,4) Then run `python conv_rate.py -f conv_test_result.csv`. Results for hex element, order 1,2,3,4
```
   Number of element/direction  convergence order of u  convergence order of p
0                            4                0.000000                     0.0
1                            6                1.812415                     NaN
2                            8                1.929877                     NaN
3                           10                1.965307                     NaN
4                           12                1.979913                     NaN

   Number of element/direction  convergence order of u  convergence order of p
0                            4                0.000000                     0.0
1                            6                2.998729                     NaN
2                            8                2.988485                     NaN
3                           10                2.986055                     NaN
4                           12                2.986057                     NaN

   Number of element/direction  convergence order of u  convergence order of p
0                            4                0.000000                     0.0
1                            6                3.996939                     NaN
2                            8                4.000354                     NaN
3                           10                4.000964                     NaN
4                           12                4.000969                     NaN

   Number of element/direction  convergence order of u  convergence order of p
0                            4                0.000000                     0.0
1                            6                4.985871                     NaN
2                            8                4.991083                     NaN
3                           10                4.993449                     NaN
4                           12                4.994788                     NaN
```

Reply:
Is pressure convergence not calculated or somehow not converging?

Reply:
That is elasticity example and pressure is not computed. For mixed Q2-Q1 we have
```
   Number of element/direction  convergence order of u  convergence order of p
0                            4                0.000000                0.000000
1                            6                3.151930                2.076015
2                            8                3.059559                2.041752
3                           10                3.025034                2.026101
```